lib/manager/composer/artifacts.ts

```diff
@@ -72,14 +73,14 @@ function getAuthJson(): string | null {
     };
   }
 
-  const gitlabCredentials = hostRules.find({
+  const [gitlabCredentials] = hostRules.findAll({
     hostType: PLATFORM_TYPE_GITLAB,
-    url: 'https://gitlab.com/api/v4/',
   });
   if (gitlabCredentials?.token) {
-    authJson['gitlab-token'] = {
-      'gitlab.com': gitlabCredentials.token,
-    };
+    const host = gitlabCredentials.hostName || 'gitlab.com';
+    authJson['gitlab-token'] = {};
+    authJson['gitlab-token'][host] = gitlabCredentials.token;
+    authJson['gitlab-domains'] = [host];
```

Comment:
Can we add a link to some public documentation about these fields?

Reply:
Hey Rhys, sorry the late reply.
For self hosted gitlab instances, the property `gitlab-domains` should contain the URL of it.

https://getcomposer.org/doc/articles/authentication-for-private-packages.md#gitlab-token